Assets/Scripts/Puzzles/MazePuzzle/MazePlayerMovement.cs

```diff
@@ -0,0 +1,131 @@
+﻿using DN.Puzzle.Maze.UI;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DN.Puzzle.Maze
+{
+	/// <summary>
+	/// Class for player movement in maze puzzle.
+	/// </summary>
+	public class MazePlayerMovement : MonoBehaviour
+	{
+		public Vector2 currentPosition { get; private set; }
+		public event Action LoseLifeEvent;
+		public event Action WinEvent;
+		private MazeBlocks[][] level;
+		private Vector2 targetPosition = Vector2.zero;
+		private Quaternion targetRotation = Quaternion.identity;
+		private List<MazeFunctions> functionQueue;
+		private float time = 0.5f;
+		private int direction = 0;
+		private float zRotation = 0;
```

Comment:
je hoeft voor al deze (behalve) time trouwens geen `0` of `Vector2.zero` in te voeren, dit zijn structs dus ze zijn automatisch al zero. Maakt het wat leesbaarder I guess, hoeft niet tho als je het fijner vindt.